go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -0,0 +1,239 @@
+package apbroute
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	adminpolicybasedrouteapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1"
+	adminpolicybasedroutelisters "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1/apis/listers/adminpolicybasedroute/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type gatewayInfoList []*gatewayInfo
+
+func (g gatewayInfoList) String() string {
+
+	s := strings.Builder{}
+	for _, item := range g {
+		s.WriteString(fmt.Sprintf("%s, ", item.gws))
+	}
+	return s.String()
+}
+
+func (g gatewayInfoList) HasIP(ip string) bool {
+	for _, i := range g {
+		if i.gws.Has(ip) {
+			return true
+		}
+	}
+	return false
+}
+
+func (g gatewayInfoList) Insert(items ...*gatewayInfo) gatewayInfoList {
+	for _, item := range items {
+		for _, ip := range item.gws.UnsortedList() {
+			if g.HasIP(ip) {
+				klog.Warningf("Duplicated static IP %s", ip)
```

Comment:
there is an assumption here of "static" but nothing about this function says this only inserts static ips?